cypress/e2e/facility_spec/FacilityHomepage.cy.ts

```diff
@@ -181,6 +221,99 @@ describe("Facility Homepage Function", () => {
     facilityNotify.closeNotificationSlide();
     loginPage.ensureLoggedIn();
     loginPage.clickSignOutBtn();
+    loginPage.loginManuallyAsDistrictAdmin();
+    loginPage.ensureLoggedIn();
+  });
+
+  it("Verify the bed capacity badge reflection", () => {
+    manageUserPage.typeFacilitySearch(facilityWithNoAvailableBeds);
+    facilityPage.verifyFacilityBadgeContent(facilityWithNoAvailableBeds);
+    manageUserPage.assertFacilityInCard(facilityWithNoAvailableBeds);
+    facilityHome.verifyOccupancyBadgeVisibility();
+    manageUserPage.assertFacilityBadgeContent("0", "0");
+
+    // link dummy doctor 12 to the facility
+    cy.awaitUrl("/users");
+    userPage.typeInSearchInput(usernameToLinkFacilityWithNoBeds);
+    userPage.checkUsernameText(usernameToLinkFacilityWithNoBeds);
+
+    cy.get("#home_facility").then(($homeFacility) => {
+      const homeFacilityText = $homeFacility.text().trim();
+      if (homeFacilityText.includes("No Home Facility")) {
+        // Link facility if no home facility exists
+        manageUserPage.clickFacilitiesTab();
+        manageUserPage.selectFacilityFromDropdown(facilityWithNoAvailableBeds);
+        manageUserPage.clickLinkFacility();
+        manageUserPage.clickHomeFacilityIcon();
+        manageUserPage.assertnotLinkedFacility(facilityWithNoAvailableBeds);
+        manageUserPage.assertHomeFacilitylink(facilityWithNoAvailableBeds);
+        manageUserPage.clickCloseSlideOver();
+      } else {
+        // Assert if facility is already linked
+        manageUserPage.assertHomeFacility(facilityWithNoAvailableBeds);
+      }
+    });
+
+    // create a new patient in the facility
+    cy.visit("/patients");
+    patientPage.createPatientWithData(newPatientData);
+    // navigate to facility page and verify the occupancy badge
+    cy.visit("/facility");
+    manageUserPage.typeFacilitySearch(facilityWithNoAvailableBeds);
+    facilityPage.verifyFacilityBadgeContent(facilityWithNoAvailableBeds);
+    facilityHome.verifyOccupancyBadgeVisibility();
+    manageUserPage.assertFacilityBadgeContent("1", "0");
+    manageUserPage.assertFacilityBadgeBackgroundColor("rgb(239, 68, 68)");
+    // create a new location and add a bed to the facility
+    facilityPage.visitAlreadyCreatedFacility();
```

Comment:
Explore the existing function for location management page loading : facilityLocation.loadLocationManagementPage or anything else exist